x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/WaterfallItem.tsx

```diff
@@ -231,6 +232,7 @@ export function WaterfallItem({
           </ErrorOverviewLink>
         ) : null}
         <Duration item={item} />
+        {item.docType === 'span' && <SyncBadge sync={item.doc.span.sync} />}
```

Comment:
since the undefined case is handled in the component, could replace this line with:
`<SyncBadge sync={item.doc?.span?.sync} />`